src/plugins/irc-events/link.js

```diff
@@ -159,7 +159,14 @@ function fetch(uri, cb) {
 	var limit = Helper.config.prefetchMaxImageSize * 1024;
 	req
 		.on("response", function(res) {
-			if (!(/^image\/.+/.test(res.headers["content-type"]))) {
+			if (/^image\/.+/.test(res.headers["content-type"])) {
+				// response is an image
+				// if Content-Length header reports a size exceeding the prefetch limit, abort fetch
+				const contentLength = parseInt(res.headers["content-length"], 10) || 0;
```

Comment:
I know you guys had some discussion about this so sorry if I missed something, but it looks to me that you don't even need `|| 0` (which I find more misleading than explicit):

```js
> const limit = 2048
undefined
> parseInt({}["content-length"], 10)
NaN
> NaN > limit
false
> null > limit
false
> undefined > limit
false
> 4096 > limit
true
```

So whatever `parseInt` returns, if the value is not an integer, the check below will be `false`.

Reply:
It's not misleading as it will force an integer type.

Reply:
It's not necessary, I don't think we should do it.

Reply:
I don't like comparing the wrong data type if I can avoid it. Passing NaN into anything number-related feels kinda dirty…

Reply:
Fair enough, I guess.